fw/websocket.c

```diff
@@ -0,0 +1,356 @@
+/**
+ *		Tempesta FW
+ *
+ * Websocket proxy protocol implementation for Tempesta FW.
+ *
+ * Copyright (C) 2022 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#undef DEBUG
+#if DBG_WS > 0
+#define DEBUG DBG_WS
+#endif
+
+#include "cfg.h"
+#include "connection.h"
+#include "websocket.h"
+#include "http.h"
+#include "http_msg.h"
```

Comment:
Now we don't need the HTTP headers:

```suggestion
#include "server.h"
```